scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -309,284 +310,453 @@ Note that while we applied this technique to assign cell types using the ADT dat
+# Bioconductor "Hub" packages provide the option to cache
+# downloads, but the interactive prompt can be annoying
+# when working with notebooks.
+# These options disable the prompt by giving permission 
+# to create the cache automatically
 ExperimentHub::setExperimentHubOption("ASK", FALSE)
 AnnotationHub::setAnnotationHubOption("ASK", FALSE)
 
-# Get Data from the Blueprint & Encode projects from celldex
-blueprint_encode <- celldex::BlueprintEncodeData(ensembl = TRUE)
+# Get Monaco 2019 data from celldex with Ensembl ids.
+monaco_ref <- celldex::MonacoImmuneData(ensembl = TRUE)
 ```
-What is this `blueprint_encode` object?
+What is this `monaco_ref` object?
 
-```{r}
-blueprint_encode
+```{r explore ref, live = TRUE}
+monaco_ref
 ```
 
-A `SummarizedExperiment` is very similar to a `SingleCellExperiment`, except rather than having each column as a cell, each column is a sample.
-Otherwise, the components are very similar: each row is still a gene, for example, and additional data about the samples
-are stored in the `colData`.
-In fact, the `SingleCellExperiment` is derived from a `SummarizedExperiment`, with some extra slots that are more relevant to single cell data.
+A `SummarizedExperiment` is very similar to a `SingleCellExperiment`, except rather than having one column per cell, each column is a *sample*.
+Otherwise, the components are very similar: each row is still a gene, for example, and additional data about the samples are stored in the `colData`.
+In fact, the `SingleCellExperiment` object is derived from a `SummarizedExperiment`, with some extra slots that are more relevant to single-cell data.
 
 What information do we have for the samples?
 
-```{r}
-colData(blueprint_encode)
+```{r explore reference sample data}
+colData(monaco_ref)
 ```
 
-Let's see how many samples of each type we have:
+There are three main columns for the sample data:
+
+- `label.main` is a more general cell type assignment.  
+
+- `label.fine` is a fine-level cell type with more specific labels.
+The exact level of granularity of these `main` and `fine` designations (and indeed the label names themselves) will vary among datasets, so it is important to look at the reference to see whether it is suitable for your application.
 
-```{r}
-data.frame(colData(blueprint_encode)) %>% 
+- `label.ont` is a standardized [Cell Ontology](https://www.ebi.ac.uk/ols/ontologies/cl) identifier. 
+Using the cell ontology can allow for more complex representations of the relationships among different cell types, but investigating that is beyond the scope of this workshop. 
+
+Another component we would like to explore is how many of each of these cell types we have in the dataset. 
+A bit of quick `dplyr` wrangling can give us the answer.
+
+```{r count cell types}
+colData(monaco_ref) |> 
+  as.data.frame() |>
   dplyr::count(label.main, label.fine)
 ```
 
+This is pretty good! 
+Most cell types have 4 replicates, which is more replicates than we often find.
```

Comment:
Not sure about the word "replicate." 
"occurrence"? Or just something like, "we see most cell types at least 4x, which is good"

Reply:
Why not? These are biological replicates!